app/src/main/java/com/vestrel00/daggerbutterknifemvp/ui/common/BaseFragmentModule.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 Vandolf Estrellado
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.vestrel00.daggerbutterknifemvp.ui.common;
+
+import android.app.Fragment;
+import android.app.FragmentManager;
+
+import com.vestrel00.daggerbutterknifemvp.inject.PerFragment;
+
+import javax.inject.Named;
+
+import dagger.Module;
+import dagger.Provides;
+
+/**
+ * Provides base fragment dependencies. This must be included in all fragment modules, which must
+ * provide a concrete implementation of {@link Fragment} and named {@link #FRAGMENT}.
+ */
+@Module
+public abstract class BaseFragmentModule {
+
+    /**
+     * See {@link BaseChildFragmentModule} class documentation regarding the need for this name.
+     */
+    public static final String FRAGMENT = "BaseFragmentModule.fragment";
+
+    static final String CHILD_FRAGMENT_MANAGER = "BaseFragmentModule.childFragmentManager";
+
+    @Provides
+    @Named(CHILD_FRAGMENT_MANAGER)
+    @PerFragment
+    static FragmentManager childFragmentManager(@Named(FRAGMENT) Fragment fragment) {
+        return fragment.getChildFragmentManager();
```

Comment:
[`Fragment.getChildFragmentManager()`](https://developer.android.com/reference/android/app/Fragment.html#getChildFragmentManager()) is available since API 17. This is one of the reasons why the non-support setup in the **master** branch uses `minSdkVersion` of 17 and not lower.

For support for API levels below 17 down to 14, take a look at the [**master-support**](https://github.com/vestrel00/android-dagger-butterknife-mvp/tree/master-support) branch.